fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/RollCallRepository.kt

```diff
@@ -0,0 +1,281 @@
+package com.github.dedis.popstellar.repository
+
+import android.app.Activity
+import android.app.Application
+import androidx.lifecycle.Lifecycle
+import com.github.dedis.popstellar.model.objects.RollCall
+import com.github.dedis.popstellar.model.objects.security.PublicKey
+import com.github.dedis.popstellar.repository.database.AppDatabase
+import com.github.dedis.popstellar.repository.database.event.rollcall.RollCallDao
+import com.github.dedis.popstellar.repository.database.event.rollcall.RollCallEntity
+import com.github.dedis.popstellar.utility.ActivityUtils.buildLifecycleCallback
+import com.github.dedis.popstellar.utility.error.UnknownRollCallException
+import com.github.dedis.popstellar.utility.error.keys.NoRollCallException
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import java.util.Collections
+import java.util.EnumMap
+import java.util.concurrent.ConcurrentHashMap
+import java.util.function.Consumer
+import java.util.stream.Collectors
+import javax.inject.Inject
+import javax.inject.Singleton
+import timber.log.Timber
+
+/**
+ * This class is the repository of the roll call events
+ *
+ * Its main purpose is to store roll calls and publish updates
+ */
+@Singleton
+class RollCallRepository @Inject constructor(appDatabase: AppDatabase, application: Application) {
+  private val rollCallsByLao: MutableMap<String, LaoRollCalls> = HashMap()
+  private val rollCallDao: RollCallDao
+  private val disposables = CompositeDisposable()
+
+  init {
+    rollCallDao = appDatabase.rollCallDao()
+    val consumerMap: MutableMap<Lifecycle.Event, Consumer<Activity>> =
+        EnumMap(Lifecycle.Event::class.java)
+    consumerMap[Lifecycle.Event.ON_STOP] = Consumer { disposables.clear() }
+    application.registerActivityLifecycleCallbacks(buildLifecycleCallback(consumerMap))
+  }
+
+  /**
+   * This either updates the roll call in the repository or adds it if absent.
+   *
+   * @param rollCall the roll call to update/add
+   */
+  fun updateRollCall(laoId: String, rollCall: RollCall) {
+    Timber.tag(TAG).d("Updating roll call on lao %s : %s", laoId, rollCall)
+
+    // Persist the rollcall
+    disposables.add(
+        rollCallDao
+            .insert(RollCallEntity(laoId, rollCall))
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+              Timber.tag(TAG).d("Successfully persisted rollcall %s", rollCall.persistentId)
+            }) { err: Throwable ->
+              Timber.tag(TAG).e(err, "Error in persisting rollcall %s", rollCall.persistentId)
+            })
```

Comment:
Are you sure this syntax is correct ? 
Isn't the second `{}` block supposed to be part of `subscribe` ?

Reply:
There's no difference, it's a trailing lambda. In short, if the last param of a function is a lambda, you can move it outside the parenthesis.
Here's some more context in a [StackOverflow post](https://stackoverflow.com/questions/60242136/rx-java-whats-the-difference-between-subscribe-vs-subscribe)